algoliasearch-apache-uber/pom.xml

```diff
@@ -44,8 +44,13 @@
                     <instructions>
                         <Bundle-SymbolicName>com.algolia.algoliasearch-apache-uber</Bundle-SymbolicName>
                         <Bundle-Name>algoliasearch-apache-uber</Bundle-Name>
-                        <Export-Package>com.algolia.search</Export-Package>
-                        <Import-Package>javax.net.ssl,javax.naming,*</Import-Package>
+                        <Bundle-Version>${project.version}</Bundle-Version>
+                        <Export-Package>com.algolia.search.*,com.algolia.search.org.apache.http.impl.nio.client</Export-Package>
```

Comment:
Maybe I am missing something but:

`com.algolia.search.org.apache.http.impl.nio.client` falls under `com.algolia.search.*` right?
Why do we need to specify `http.impl.nio...`?

Reply:
Precisely: because bundling (hence exploration/resolution/expansion of `com.algolia.search.*`) happens _before_ shading (hence before org.apache... becomes com.algolia.search.org.apache...). And (answering your second question) this is also why the full path is specified (`com.algolia.search.org.apache.http.impl.nio.client`): the bundling plugin can't find it at all at the time it runs (it is nice enough to add it to the MANIFEST Export-Package section nonetheless)

Reply:
@aseure here is what I was suggesting last time we talk, to explicit the relationship between this export and the shading below:

Add a property in a `<properties>` section above, eg:
```
<properties>
  <relocated.org.apache>com.algolia.search.org.apache</relocated.org.apache>
</properties>
```

Leverage that property in the configuration of the bundling package:
```
<Export-Package>com.algolia.search.*,${relocated.org.apache}.http.impl.nio.client</Export-Package>
```

Leverage that same property in the configuration of the shading plugin, in the `<relocation>` section below:
```
<pattern>org.apache</pattern>
<shadedPattern>${relocated.org.apache}</shadedPattern>
```

That way, it is made explicit, by code, that there is a relationship between the two.

Reply:
@BenoitPerrot Thanks for all the answers you gave to @Ant-hem and for the recommendation to add the property. This is indeed a very good idea! Just adding it right now.